### ISSUE TITLE
Strange (bad) interaction with LaTeX buffers

### ISSUE THREAD

**Issue by Rmano** [davidhalter/jedi-vim#440]
Hi --- 
I have installed `jedi-vim` and I find it really nice to edit python files. But I came across a problem that I am unable to solve --- it makes editing of LaTeX files much more complex. The problem is the following. I have in my `.vimrc`: 

```
Plug 'davidhalter/jedi-vim'
    autocmd FileType python setlocal completeopt-=preview 
```

And when I edit a LaTeX file, the `$...$` marks that define an inline math disappears unless the line is under the cursor. See: 

Case 1, cursor in different line of the math formula (the cursor does not appear on screenshot, but believe me...) 

![selection_036](https://cloud.githubusercontent.com/assets/6414907/8773950/d84d6838-2ed8-11e5-80ef-c24c3814cbfc.png)

Case 2, few instant after, cursor in the same line: 

![selection_037](https://cloud.githubusercontent.com/assets/6414907/8773962/f4f523d6-2ed8-11e5-9d02-97d91bff34ac.png)

This make LaTeX files almost impossible to edit. Is there a way to restrict all of the `jedi-vim` effects to python buffers?

PD I checked that commenting out the plugin will remove the problem.

**Comment by Rmano** [comment]
The problem (see http://vi.stackexchange.com/questions/3999/jedi-vim-and-vim-latex-buffers-problems ) seems to be that --- I do not know why --- the `set conceallevel=2` get triggered for LaTeX buffers. Adding

```
autocmd FileType tex set concealleval=0 
```
 fixes the issue. Still, I am quite confused --- maybe a strange interaction with my `.vimrc`? Puzzled.

**Comment by blueyed** [comment]
I am re-opening the issue.  There's likely something that jedi-vim could do better for this case, see http://vi.stackexchange.com/questions/3999/jedi-vim-and-vim-latex-buffers-problems#comment5775_4001 and the other comments.

**Comment by Rmano** [comment]
I am not sure, but I think that the problem is (could be) that `vimtex` loads the syntax file for sub-syntax formatting (as in `listing` environments containing python code). See around line 66 in https://github.com/lervag/vimtex/blob/master/after/syntax/tex.vim 

Probably the only solution is that the `set conceallevel` should be done in another place, not in the syntax definition file --- but do not trust me, I am a newbie in this.

**Comment by blueyed** [comment]
> Probably the only solution is that the set conceallevel should be done in another place

Yeah, that's what I'd say, too - probably in the the filetype plugin instead.

**Comment by davidhalter** [comment]
> It's tied to setting the conceil syntax, and maybe could be done when setting up / tearing down the call signatures.

True. However, I'm not sure if it's necessary.

**Comment by blueyed** [comment]
There is better support for setting up call signatures in #652, e.g. not setting `conceallevel` at all when using another mode than `1`.

**Comment by ruestefa** [comment]
More than 7 years later, I've just experienced the very same thing in a JSON file that I opened in the same Neovim instance as a Python file:

```
  conceallevel=2                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    
        Last set from ~/.local/share/nvim/site/pack/packer/opt/jedi-vim/after/syntax/python.vim line 12
```

Version: `344814c214391368674813e3a11143448e73e749`
